src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -516,75 +545,127 @@ private void computeVisibleFieldsAndInnerClasses(LocalMemberTable lmt, Kind kind
+        //
         // nb. This statement has side effects that can initialize
         // members of the overriddenMethodTable field, so it must be
         // evaluated eagerly with toList().
-        List<Element> inheritedMethodsList = inheritedMethods.stream()
+        List<Element> inheritedMethods = parentMethods.stream()
                 .filter(e -> allowInheritedMethod((ExecutableElement) e, overriddenByTable, lmt))
                 .toList();
 
-        // Filter out the local methods, that do not override or simply
-        // overrides a super method, or those methods that should not
-        // be visible.
-        Predicate<ExecutableElement> isVisible = m -> {
-            OverriddenMethodInfo p = overriddenMethodTable.getOrDefault(m, null);
-            return p == null || !p.simpleOverride;
+        // filter out "simple overrides" from local methods
+        Predicate<ExecutableElement> nonSimpleOverride = m -> {
+            OverrideInfo i = overriddenMethodTable.get(m);
+            return i == null || !i.simpleOverride;
         };
 
         Stream<ExecutableElement> localStream = lmt.getOrderedMembers(Kind.METHODS)
                 .stream()
                 .map(m -> (ExecutableElement)m)
-                .filter(isVisible);
+                .filter(nonSimpleOverride);
 
         // Merge the above list and stream, making sure the local methods precede the others
         // Final filtration of elements
-        List<Element> list = Stream.concat(localStream, inheritedMethodsList.stream())
+        // FIXME add a test to assert the order or remove that part of the comment above ^
+        List<Element> list = Stream.concat(localStream, inheritedMethods.stream())
                 .filter(this::mustDocument)
                 .toList();
 
         visibleMembers.put(Kind.METHODS, list);
 
-        // Copy over overridden tables from the lineage, and finish up.
+        // copy over overridden tables from the lineage
         for (VisibleMemberTable pvmt : parents) {
+            // a key in overriddenMethodTable is a method _declared_ in the respective parent;
+            // no two _different_ parents can share a declared method, by definition;
+            // if parents in the list are different (i.e. the list of parents doesn't contain duplicates),
+            //   then no keys are equal and thus no replace happens
+            // if the list of parents contains duplicates, values for the equal keys are equal,
+            //   so no harm if they are replaced in the map
+            assert putAllIsNonReplacing(overriddenMethodTable, pvmt.overriddenMethodTable);
             overriddenMethodTable.putAll(pvmt.overriddenMethodTable);
         }
     }
 
-    boolean isEnclosureInterface(Element e) {
-        TypeElement enclosing = utils.getEnclosingTypeElement(e);
-        return utils.isPlainInterface(enclosing);
+    private static <K, V> boolean putAllIsNonReplacing(Map<K, V> dst, Map<K, V> src) {
```

Comment:
naming suggestion:   the dominant initial word here is `put`  but the method is a predicate, suggesting that the initial word should be `is`